cmd/satellite/gracefulexit.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"fmt"
+	"io"
+	"os"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/satellite/satellitedb"
+)
+
+// generateGracefulExitCSV creates a report with graceful exit data for exiting or exited nodes in a given period
+func generateGracefulExitCSV(ctx context.Context, start time.Time, end time.Time, output io.Writer) error {
+	db, err := satellitedb.New(zap.L().Named("db"), gracefulExitCfg.Database)
+	if err != nil {
+		return errs.New("error connecting to master database on satellite: %+v", err)
+	}
+	defer func() {
+		err = errs.Combine(err, db.Close())
+	}()
+
+	// TODO: add overlay function for getting exiting/exited nodes in timespan
+	// TODO: add a config param similar to storagenode/retain config.Status for choosing between all/exiting/exited
+	nodeIDs, err := db.OverlayCache().GetGracefulExitNodes(ctx, false, start, end)
+	if err != nil {
+		return err
+	}
+
+	w := csv.NewWriter(output)
+	headers := []string{
+		"nodeID",
+		"walletAddress",
+		"nodeCreationDate",
+		"initiatedGracefulExit",
+		"completedGracefulExit",
+		"transferredGB",
+	}
+	if err := w.Write(headers); err != nil {
+		return err
+	}
+
+	for _, id := range nodeIDs {
+		node, err := db.OverlayCache().Get(ctx, id)
+		if err != nil {
+			return err
+		}
+		exitStatus, err := db.OverlayCache().GetExitStatus(ctx, id)
+		if err != nil {
+			return err
+		}
+
+		nextRow := []string{
+			node.Id.String(),
+			node.Operator.Wallet,
+			nil, // TODO how to get creation date
```

Comment:
wait what's the problem with getting a creation date or amount transferred? do those columns need to be added to the nodes table also?

Reply:
I didn't see them on the `node` variable here. I think they might need to be added to the nodes table, or I need to find some other way to access them.

Reply:
creation date (i assume this joined date) in `nodes.created_at` i thought
bytes transferred is stored in `graceful_exit_progress`

Reply:
I had to add creation date to the node dossier. I was able to get the bytes transferred from graceful exit progress. Thanks Ethan!